src/modules/useSupersetEmbeddedDashboardFieldsState.js

```diff
@@ -0,0 +1,80 @@
+import { useCallback, useReducer } from 'react'
+
+export const fieldNames = {
+    title: 'title',
+    code: 'code',
+    description: 'description',
+    supersetEmbedId: 'supersetEmbedId',
+    showChartControls: 'showChartControls',
+    showFilters: 'showFilters',
+}
+const defaultInitialValues = {
+    [fieldNames.title]: '',
+    [fieldNames.code]: '',
+    [fieldNames.description]: '',
+    [fieldNames.supersetEmbedId]: '',
+    [fieldNames.showChartControls]: true,
+    [fieldNames.showFilters]: true,
+}
+
+const FIELD_CHANGE = 'FIELD_CHANGE'
+const SUPERSET_FIELD_BLUR = 'SUPERSET_FIELD_BLUR'
+const UUID_PATTERN =
+    /^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/i
```

Comment:
There's not some tiny 3rd party library that could validate the uuid instead? If we intend to use our own like this, then there should be unit tests for it.